cmd/tools/asup/asup.go

```diff
@@ -85,76 +84,91 @@ func DoAsupMessage(config string, collectors []collector.Collector, status *matr
 
 	connection := getdata(config, pollerName)
 
-	if msg, err = buildAsupMessage(collectors, status, connection, harvestUUID); err != nil {
-		return errors.New(errors.ERR_CONFIG, "failed to build ASUP message")
+	if msg, err = buildAsupMessage(collectors, status, connection); err != nil {
+		return fmt.Errorf("failed to build ASUP message poller:%s %w", pollerName, err)
 	}
 
 	if err = sendAsupMessage(msg, pollerName); err != nil {
-		return errors.New(errors.ERR_CONFIG, "failed to send ASUP message")
+		return fmt.Errorf("failed to send ASUP message poller:%s %w", pollerName, err)
 	}
 
 	return nil
 }
 
-// This function would be used to invoke harvest-asup(private repo)
+// This function forks the autosupport binary
 func sendAsupMessage(msg *asupMessage, pollerName string) error {
+	err := sendAsupVia(msg, pollerName, "./bin/asup")
+	if errors.Is(err, os.ErrNotExist) {
+		err = sendAsupVia(msg, pollerName, "../harvest-private/harvest-asup/bin/asup")
+	}
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func sendAsupVia(msg *asupMessage, pollerName string, asupExecPath string) error {
 	var (
 		payloadPath string
 		err         error
 	)
 	asupTimeOutLimit := 10 * time.Second
```

Comment:
asup exec timeout is also need to decide may be 5 mins ?

Reply:
how does exec timeout differ from `asupTimeOutLimit`

Reply:
actually this is the exec timeout we have provided in context at line 135